src/commcare_cloud/terraform/modules/logshipping/check_file_lambda.py

```diff
@@ -0,0 +1,113 @@
+    response_query_execution_id = client.start_query_execution(
+        QueryString = params['query'],
+        QueryExecutionContext = {
+            'Database' : "default"
+        },
+        ResultConfiguration = {
+            'OutputLocation': 's3://' + params['bucket'] + '/' + params['path']
+        }
+    )
+
+    if not wait:
+        return response_query_execution_id['QueryExecutionId']
+    else:
+        response_get_query_details = client.get_query_execution(
+            QueryExecutionId = response_query_execution_id['QueryExecutionId']
+        )
+        status = 'RUNNING'
+        iterations = 360 # 30 mins
+
+        while (iterations > 0):
+            iterations = iterations - 1
+            response_get_query_details = client.get_query_execution(
+            QueryExecutionId = response_query_execution_id['QueryExecutionId']
+            )
+            res = response_get_query_details['QueryExecution']
+            status = response_get_query_details['QueryExecution']['Status']['State']
+            print(status)
+            if (status == 'FAILED') or (status == 'CANCELLED') :
+                response = response_get_query_details['QueryExecution']['Status']['StateChangeReason']
+                return status, response
+
+            elif status == 'SUCCEEDED':
+                location = response_get_query_details['QueryExecution']['ResultConfiguration']['OutputLocation']
+
+                ## Function to get output results
+                response_query_result = client.get_query_results(
+                    QueryExecutionId = response_query_execution_id['QueryExecutionId']
+                )
+                result_data = response_query_result['ResultSet']
+
+                if len(response_query_result['ResultSet']['Rows']) > 1:
+                    header = response_query_result['ResultSet']['Rows'][0]
+                    rows = response_query_result['ResultSet']['Rows'][1:]
+                    return location, status
+                else:
+                    return location, "No rows"
+            else:
+                time.sleep(5)
+    return False
+
+def handler(event, context=None, params=None):
+    # get UTC date 
+    utc_datetime = datetime.utcnow()
+    year = utc_datetime.strftime('%Y')
+    month = utc_datetime.strftime('%m')
+    day = utc_datetime.strftime("%d")
+    hour = utc_datetime.strftime('%H')
+
+    DATABASE = 'default'
+    TABLE = 'formplayer_request_response_logs'
+
+    # S3 constant
+    S3_OUTPUT = 's3://dimagi-commcare-staging-logs'
+    S3_BUCKET = 'dimagi-commcare-staging-logs'
+    STREAM_PATH='formplayer-request-response-logs-partitioned-staging'
+    verifyKey = f"{STREAM_PATH}/year={year}/month={month}/day={day}/hour={hour}"
+
+    addPartitionQuery=f"ALTER TABLE {TABLE} ADD PARTITION (year={year}, month={month}, day={day}, hour={hour}) LOCATION 's3://{S3_BUCKET}/{STREAM_PATH}/year={year}/month={month}/day={day}/hour={hour}/';"
```

Comment:
This will only add the partition for the current hour, so if the lambda function for some reason doesn't fire one hour (perhaps an error of some sort) then the partitions for those hours will silently never get added, right? Can this run `MSCK REPAIR {TABLE}` instead? Is that too expensive or something?

Reply:
It is too expensive Danny. It will scan all files and adds partitions.